### PR TITLE
HPCC-29654 Add manifest Support to NLP plugin

### PR DESCRIPTION
**Pull request by GordonSmith**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [x] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Review by dehilsterlexis — APPROVED** [review]
I have tested this with several analyzers and it is working.

File: .gitignore [hunk 1]
```diff
@@ -17,4 +17,4 @@ eclcc.log
 ./helm/examples/azure/sa/env-sa
 ./dockerfiles/platform-build-incremental/hpcc.gitpatch
 .env
-vcpkg.json
+/vcpkg.json
```
Comment (ghalliday) [inline .gitignore:20]: is this change correct?  Seems and unlikely path.

File: plugins/nlp/manifest.cpp [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+            manifest.setown(createPTree());
+        }
+    }
+
+public:
+    Manifest(const Manifest &) = delete;
+    Manifest &operator=(const Manifest &) = delete;
+    ~Manifest()
+    {
+        DBGLOG("MANIFEST:  Destructor");
```
Comment (ghalliday) [inline plugins/nlp/manifest.cpp:72]: Remove now?

File: plugins/nlp/manifest.cpp [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+
+    static IConstWorkUnit *getWorkunit(ICodeContext *ctx)
+    {
+        Owned<IWorkUnitFactory> factory = getWorkUnitFactory();
+        StringAttr wuid;
+        wuid.setown(ctx->getWuid());
+        return factory->openWorkUnit(wuid);
+    }
+
+    Manifest(ICodeContext *ctx) : cw(getWorkunit(ctx))
```
Comment (ghalliday) [inline plugins/nlp/manifest.cpp:30]: This will leak - cw needs to be Owned\<\>

File: plugins/nlp/manifest.cpp [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+    {
+        DBGLOG("MANIFEST:  Constructor");
```
Comment (ghalliday) [inline plugins/nlp/manifest.cpp:32]: remove?

File: plugins/nlp/manifest.cpp [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+    }
+
+    static std::shared_ptr<IManifest> getInstance(ICodeContext *ctx)
+    {
+        std::call_once(initFlag, [ctx]()
+                       { instance = std::shared_ptr<Manifest>(new Manifest(ctx)); });
+        return instance;
+    }
+
+    const char *stripLeadingCharacters(const char *input) const
```
Comment (ghalliday) [inline plugins/nlp/manifest.cpp:82]: function name doesn't really tell you what it is stripping.

File: plugins/nlp/manifest.cpp [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+            return true;
+        }
+        return false;
+    }
+
+    IPropertyTree *queryResourcePT(const char *_path) const
+    {
+        const char *path = stripLeadingCharacters(_path);
+        StringBuffer xpath;
+        if (!manifestDir.length())
```
Comment (ghalliday) [inline plugins/nlp/manifest.cpp:125]: trivial: manifestDir.isEmpty()

File: vcpkg-linux.code-workspace [hunk 7]
```diff
@@ -14,14 +14,14 @@
 			"-DSUPPRESS_V8EMBED=ON",
 			"-DSUPPRESS_REMBED=ON"
 		],
-		"editor.formatOnSave": false
```
Comment (ghalliday) [inline vcpkg-linux.code-workspace:17]: deliberate?

**Review by ghalliday — COMMENTED** [review]
Some initial comments

File: .gitignore (hunk 1 above)
Comment (GordonSmith) [reply .gitignore:20]: It is specifically for the generated vcpkg.json in the route path - others can exist in the vcpkg_overlays which we would want to keep.

File: vcpkg-linux.code-workspace (hunk 7 above)
Comment (GordonSmith) [reply vcpkg-linux.code-workspace:17]: No.

**Comment by GordonSmith** [comment]
@ghalliday changes made as requested.

**Pull request merged**